src/components/articles/ArticleVote.jsx

```diff
@@ -0,0 +1,48 @@
+import { useState } from "react";
+import { patchArticleVotes } from "../../utils/api";
+import ErrorDisplay from "../minor/ErrorDisplay";
+
+const ArticleVote = ({ article_id, votes, setArticle }) => {
+  const [voteChange, setVoteChange] = useState(0);
+  const [error, setError] = useState(null);
+
+  const handleVote = async (increment) => {
+    setVoteChange((currentChange) => currentChange + increment);
+    setError(null);
+
+    try {
+      const updatedArticle = await patchArticleVotes(article_id, increment);
+      setArticle((article) => ({
+        ...article,
```

Comment:
I would rather we didn't setArticle here. The idea of this component is that we only trigger a re-render on the votes number rather than the whole article.